### ISSUE TITLE
Cannot Restart Proxy

### ISSUE THREAD

**Issue by MisterAcoustic** [justcoding121/Titanium-Web-Proxy#193]
Hello,

I am trying to 'restart' the proxy, but calling Stop then Start doesn't seem to work.

I have tried calling my StopProxy () then StartProxy functions below, as well as calling StopProxy(), then proxyServer.Dispose(), then creating an entirely new Titanium proxy and starting that - but none of those things successfully capture traffic after the restart.

Am I doing something wrong, or is this a bug?

Any help is appreciated.

`

         public void StartProxy()
        {
            proxyServer.ServerCertificateValidationCallback += OnCertificateValidation;
            proxyServer.ClientCertificateSelectionCallback += OnCertificateSelection;

            var explicitEndPoint = new ExplicitProxyEndPoint(IPAddress.Any, MyPortNumber, true){};

            proxyServer.AddEndPoint(explicitEndPoint);

            foreach (var endPoint in proxyServer.ProxyEndPoints)
                Log(string.Format("Listening on '{0}' endpoint at Ip {1} and port: {2} ",
                    endPoint.GetType().Name, endPoint.IpAddress, endPoint.Port));

            proxyServer.Start();

            //Only explicit proxies can be set as system proxy!
            proxyServer.SetAsSystemHttpProxy(explicitEndPoint);
            proxyServer.SetAsSystemHttpsProxy(explicitEndPoint);
        }

        public void StopProxy()
        {
            proxyServer.BeforeRequest -= RequestHandler;
            proxyServer.BeforeResponse -= ResponseHandler;
            proxyServer.ServerCertificateValidationCallback -= OnCertificateValidation;
            proxyServer.ClientCertificateSelectionCallback -= OnCertificateSelection;

            proxyServer.Stop();
        }`

**Comment by MisterAcoustic** [comment]
Can anyone say whether they can successfully restart the proxy? It does not work for me, and I don't want my users to have to restart my application continually. Does the 'Help Wanted' tag mean that this should work, and I'm the one who needs help, or that this is broken and help is needed to fix it?

**Comment by justcoding121** [comment]
**Help wanted** would mean we need someone with time to help you to figure out the issue in detail. Prima Facie I couldn't reproduce this earlier. Could be escalated to bug at a later stage. I am sorry that I am busy with handful of personal projects at the moment. 

I tried it again without any issue. Below line is unnecessary in your code by the way. Why are we unsubscribing a handler which was never invoked?

 proxyServer.ServerCertificateValidationCallback -= OnCertificateValidation;
 proxyServer.ClientCertificateSelectionCallback -= OnCertificateSelection;

Also if your are calling stop() and start() using Console.Read() as is in the example operation, make sure the stop() is indeed getting called, and you are not calling again start() after stop().

**Comment by MisterAcoustic** [comment]
Thanks for the reply. I have both solved my problem, and decided to do things differently. This is not a bug in the Titanium Web Proxy. My mistake was changing where and how my Request and Response handlers were being installed, which resulted in those handlers not being re-installed in some circumstances. Prior to figuring that out, I decided to dispense with the idea that I needed to restart the proxy, and instead just implemented a flag that allows me to ignore traffic when I want to. I think that's a better approach for my situation anyway. 

Thanks for your help, and good luck on your projects.
